src/random.cpp

```diff
@@ -43,15 +43,21 @@ static void RandFailure()
 
 static inline int64_t GetPerformanceCounter()
 {
-    int64_t nCounter = 0;
-#ifdef WIN32
-    QueryPerformanceCounter((LARGE_INTEGER*)&nCounter);
+#if defined(_MSC_VER)
+    return __rdtsc();
+#elif defined(__i386__)
+    uint64_t r;
+    __asm__ volatile ("rdtsc" : "=A"(r));
+    return r;
+#elif defined(__x86_64__) || defined(__amd64__)
+    uint64_t r1, r2;
+    __asm__ volatile ("rdtsc" : "=a"(r1), "=d"(r2));
+    return (r2 << 32) | r1;
```

Comment:
Maybe link to https://en.wikipedia.org/wiki/Time_Stamp_Counter for readers that don't know ASM?

Also,  I couldn't find a description on the constraints `a` or `d`,  I assume they are just writing to the inline variables `r1` and `r2`,  but I found documentation on this to be difficult to find.